File: apps/backend/internal/modules/menu/model/menu.go
package model

import "gorm.io/gorm"

type Menu struct {
	gorm.Model
	Title          string `gorm:"type:varchar(64);not null;comment:菜单名称"`
	Path           string `gorm:"type:varchar(255);not null;unique;comment:前端路由路径"`
	Icon           string `gorm:"type:varchar(255);comment:图标"`
	ParentID       *uint  `gorm:"index;comment:父级菜单ID"`
	Sort           int    `gorm:"default:0;comment:排序"`
	Hidden         bool   `gorm:"default:false;comment:是否隐藏菜单"`
	PermissionCode string `gorm:"type:varchar(255);comment:权限标识符"`
	ExternalLink   bool   `gorm:"default:false;comment:是否外链"`
	KeepAlive      bool   `gorm:"default:true;comment:是否缓存页面"`
}

File: apps/web/src/api/menu.ts
import { z } from 'zod'
import { apiClient } from '../common/apiClient'

export const MenuSchema = z.object({
  id: z.number(),
  parentId: z.number().optional().nullable(),
  title: z.string().min(1, '菜单名称不能为空'),
  path: z.string().min(1, '路由路径不能为空'),
  icon: z.string().optional(),
  hidden: z.boolean().optional(),
  keepAlive: z.boolean().optional(),
  externalLink: z.boolean().optional(),
  sort: z.number().optional(),
  permissionCode: z.string().min(1, '权限码不能为空'),
})

export type MenuDTO = z.infer<typeof MenuSchema>
export const MenuSubmitSchema = MenuSchema.omit({ id: true })

export async function fetchMenus() {
  return apiClient<MenuDTO[]>('/sys/menu/list')
}

export async function createMenu(menu: MenuDTO) {
  return apiClient<MenuDTO>('/sys/menu', {
    method: 'POST',
    data: menu,
  })
}

export async function updateMenu(id: number, menu: MenuDTO) {
  return apiClient<MenuDTO>(`/sys/menu/${id}`, {
    method: 'PUT',
    data: menu,
  })
}

export async function deleteMenu(id: number) {
  return apiClient<any>(`/sys/menu/${id}`, {
    method: 'DELETE',
  })
}
